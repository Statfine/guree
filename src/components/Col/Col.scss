@import "../../common/utils/index.scss";

$total: 24;

@function columnOffset($col, $total: $total) {
  @if $col == 0 {
    @return 0;
  } @else {
    @return percentage($col/$total);
  }
}

@mixin make-grid($responsive: "") {
  @for $i from 0 to $total + 1 {
    @if $i == 0 {
      &-#{$responsive}offset-0 {
        margin-left: 0;
      }
      &-#{$responsive}order-0 {
        order: 0;
      }
      &-push-0 {
        left: auto;
      }

      &-pull-0 {
        right: auto;
      }
      &-0 {
        display: none;
      }
    } @else {
      &-#{$responsive}-offset-#{$i} {
        margin-left: columnOffset($i);
      }
      &-#{$responsive}-order-#{$i} {
        order: $i;
      }
      &-#{$responsive}-push-#{$i} {
        left: columnOffset($i);
      }
      &-#{$responsive}-pull-#{$i} {
        right: columnOffset($i);
      }
      &-#{$responsive}-#{$i} {
        display: block;
        flex: 0 0 columnOffset($i);
        max-width: columnOffset($i);
      }
    }
  }
}

.#{scopedClass(col)} {
  max-width: 100%;
  min-height: 1px;
  position: relative;
  @include make-grid();
  // xs
  @media only screen and (max-width: 576px) {
    @include make-grid("xs");
  }
  // sm
  @media only screen and (min-width: 576px) {
    @include make-grid("sm");
  }
  // md
  @media only screen and (min-width: 768px) {
    @include make-grid("md");
  }
  // lg
  @media only screen and (min-width: 992px) {
    @include make-grid("lg");
  }
  // xl
  @media only screen and (min-width: 1200px) {
    @include make-grid("xl");
  }
  // xxl
  @media only screen and (min-width: 1600px) {
    @include make-grid("xxl");
  }
}
