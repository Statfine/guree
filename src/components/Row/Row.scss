@import "../../common/utils/index.scss";

.#{scopedClass(row)} {
  display: flex;
  flex-flow: row wrap;

  &-top {
    align-items: flex-start;
  }
  &-middle {
    align-items: center;
  }
  &-bottom {
    align-items: flex-end;
  }
  &-start {
    justify-content: flex-start;
  }
  &-end {
    justify-content: flex-end;
  }
  &-center {
    justify-content: center;
  }
  &-space-around {
    justify-content: space-around;
  }
  &-space-between {
    justify-content: space-between;
  }
}
