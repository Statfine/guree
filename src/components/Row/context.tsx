import { createContext, Context } from "react";

export interface RowContextState {
  horizontal?: number;
  vertical?: number;
}

const RowContext: Context<RowContextState> = createContext({});

export default RowContext;
