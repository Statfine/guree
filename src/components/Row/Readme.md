```jsx
import Col from "@components/Col";
<Row justify="start">
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
</Row>;
```

```jsx
import Col from "@components/Col";
<Row justify="end">
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
</Row>;
```

```jsx
import Col from "@components/Col";
<Row justify="center">
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
</Row>;
```

```jsx
import Col from "@components/Col";
<Row justify="space-around">
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
</Row>;
```

```jsx
import Col from "@components/Col";
<Row justify="space-between">
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
</Row>;
```

```jsx
import Col from "@components/Col";
<Row justify="space-between">
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
</Row>;
```

```jsx
import Col from "@components/Col";
<Row gutter={16}>
  <Col className="gutter-row" span={6}>
    <div style={{ background: "#00a0e9" }}>col-6</div>
  </Col>
  <Col className="gutter-row" span={6}>
    <div style={{ background: "#00a0e9" }}>col-6</div>
  </Col>
  <Col className="gutter-row" span={6}>
    <div style={{ background: "#00a0e9" }}>col-6</div>
  </Col>
  <Col className="gutter-row" span={6}>
    <div style={{ background: "#00a0e9" }}>col-6</div>
  </Col>
</Row>;
```

```jsx
import Col from "@components/Col";
<Row gutter={{ xs: 8, sm: 16, md: 24, lg: 32 }}>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
</Row>;
```

```jsx
import Col from "@components/Col";
<Row gutter={[16, 24]}>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
</Row>;
```

```jsx
import Col from "@components/Col";
<Row
  gutter={[
    { xs: 2, sm: 4, md: 6, lg: 8, xl: 10, xxl: 12 },
    { xs: 14, sm: 16, md: 18, lg: 20, xl: 22, xxl: 24 },
  ]}
>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
</Row>;
```
