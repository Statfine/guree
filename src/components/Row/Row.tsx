import React, { ReactNode, useState, CSSProperties } from "react";
import "./Row.scss";
import { combineClasses, scopedClass, classesObj } from "@utils/index";
import RowContext from "@components/Row/context";

type GutterType =
  | number
  | {
      xs?: number;
      sm?: number;
      md?: number;
      lg?: number;
      xl?: number;
      xxl?: number;
    };

interface Props {
  children?: ReactNode;
  align?: "top" | "middle" | "bottom";
  justify?: "start" | "end" | "center" | "space-around" | "space-between";
  gutter?: GutterType | [GutterType, GutterType];
}

const Row: React.FunctionComponent<Props> = ({
  children,
  align,
  justify,
  gutter,
}: Props) => {
  const colClass = (...classes: (string | Array<string> | classesObj)[]) =>
    scopedClass("row", ...classes);

  const MediaMap: {
    [k: string]: string;
  } = {
    xs: "(max-width: 576px)",
    sm: "(min-width: 576px)",
    md: "(min-width: 768px)",
    lg: "(min-width: 992px)",
    xl: "(min-width: 1200px)",
    xxl: "(min-width: 1600px)",
  };

  const parseGutter = (gutter: GutterType) => {
    if (typeof gutter === "number") {
      return gutter;
    }
    return Object.entries(gutter).reduce((res, cur) => {
      const [mediaKey, mediaValue] = cur;
      const mql = window.matchMedia(MediaMap[mediaKey]);
      if (mql.matches) {
        return mediaValue || 0;
      }
      return res;
    }, 0);
  };

  const getGutter = (
    gutter: GutterType | [GutterType, GutterType] | undefined
  ) => {
    if (typeof gutter === "undefined") {
      return {
        horizontal: 0,
        vertical: 0,
      };
    }
    if (Array.isArray(gutter)) {
      return {
        horizontal: parseGutter(gutter[0]),
        vertical: parseGutter(gutter[1]),
      };
    }
    return {
      horizontal: parseGutter(gutter),
      vertical: 0,
    };
  };

  const [{ horizontal, vertical }, setState] = useState(getGutter(gutter));

  const responiveGutter = (gutter: GutterType, cb: Function, key: string) => {
    if (typeof gutter === "number") {
      return gutter;
    }
    Object.entries(gutter).forEach(([mediaKey, mediaValue]) => {
      const mql = window.matchMedia(MediaMap[mediaKey]);
      mql.addListener((e) => {
        if (e.matches) {
          cb({
            [key]: mediaValue || 0,
          });
        }
      });
    });
  };

  const bindGutter = (
    gutter: GutterType | [GutterType, GutterType] | undefined,
    setState: React.Dispatch<
      React.SetStateAction<{
        horizontal: number;
        vertical: number;
      }>
    >
  ) => {
    if (typeof gutter === "undefined") {
      return;
    }
    if (Array.isArray(gutter)) {
      gutter.forEach((x, i) => {
        responiveGutter(x, setState, ["horizontal", "vertical"][i]);
      });
    } else {
      responiveGutter(gutter, setState, "horizontal");
    }
  };

  bindGutter(gutter, setState);

  const styleClassNames = (align?: string, justify?: string) => {
    return combineClasses({
      [`${colClass(`${align}`)}`]: Boolean(align),
      [`${colClass(`${justify}`)}`]: Boolean(justify),
    });
  };

  const gutterStyle: CSSProperties = {
    ...(horizontal > 0
      ? {
          marginLeft: horizontal / -2,
          marginRight: horizontal / -2,
        }
      : {}),
    ...(vertical! > 0
      ? {
          marginTop: vertical / -2,
          marginBottom: vertical / 2,
        }
      : {}),
  };

  const className = combineClasses(colClass(), styleClassNames(align, justify));
  return (
    <RowContext.Provider value={{ horizontal, vertical }}>
      <div className={className} style={gutterStyle}>
        {children}
      </div>
    </RowContext.Provider>
  );
};

export default Row;
