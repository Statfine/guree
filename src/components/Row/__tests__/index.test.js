import React from "react";
import renderer from "react-test-renderer";
import Row from "..";
import Col from "@components/Col";

const screenList = [575, 576, 768, 992, 1200, 1600];
const resizeWindow = (width) => {
  window.innerWidth = width;
};
// const button = mount(<Button onClick={fn}>button</Button>);

Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => {
    return {
      matches: (() => {
        const minOrMax = query.slice(1, 4);
        const width = Number(query.slice(12).replace("px)", ""));
        const sceenWidth = window.innerWidth;
        if (minOrMax === "max") {
          return sceenWidth < width;
        } else if (minOrMax === "min") {
          return sceenWidth >= width;
        }
        return false;
      })(),
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    };
  }),
});

const numberTypeComponent = (
  <Row gutter={16}>
    <Col span={6}>col-6</Col>
    <Col span={6}>col-6</Col>
    <Col span={6}>col-6</Col>
    <Col span={6}>col-6</Col>
  </Row>
);
const objectTypeComponent = (
  <Row gutter={{ xs: 8, sm: 16, md: 24, lg: 32, xl: 64, xxl: 128 }}>
    <Col span={6}>col-6</Col>
    <Col span={6}>col-6</Col>
    <Col span={6}>col-6</Col>
    <Col span={6}>col-6</Col>
  </Row>
);
const ArrayNumberComponent = (
  <Row gutter={[16, 24]}>
    <Col span={6}>col-6</Col>
    <Col span={6}>col-6</Col>
    <Col span={6}>col-6</Col>
    <Col span={6}>col-6</Col>
  </Row>
);

const ArrayObjectComponent = (
  <Row
    gutter={[
      { xs: 2, sm: 4, md: 6, lg: 8, xl: 10, xxl: 12 },
      { xs: 14, sm: 16, md: 18, lg: 20, xl: 22, xxl: 24 },
    ]}
  >
    <Col span={6}>col-6</Col>
    <Col span={6}>col-6</Col>
    <Col span={6}>col-6</Col>
    <Col span={6}>col-6</Col>
  </Row>
);

describe("Row", () => {
  it("正确导出 Row 组件", () => {
    const row = renderer.create(<Row></Row>).toJSON();
    expect(row).toMatchSnapshot();
  });

  it("测试 Row 组件 justify 参数", () => {
    const start = renderer
      .create(
        <Row justify="start">
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
        </Row>
      )
      .toJSON();
    const end = renderer
      .create(
        <Row justify="end">
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
        </Row>
      )
      .toJSON();
    const center = renderer
      .create(
        <Row justify="center">
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
        </Row>
      )
      .toJSON();
    const spaceAround = renderer
      .create(
        <Row justify="space-around">
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
        </Row>
      )
      .toJSON();
    const spaceBetween = renderer
      .create(
        <Row justify="space-between">
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
        </Row>
      )
      .toJSON();
    expect(start).toMatchSnapshot();
    expect(end).toMatchSnapshot();
    expect(center).toMatchSnapshot();
    expect(spaceAround).toMatchSnapshot();
    expect(spaceBetween).toMatchSnapshot();
  });

  it("测试 Row 组件 align 参数", () => {
    const top = renderer
      .create(
        <Row align="top">
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
        </Row>
      )
      .toJSON();
    const middle = renderer
      .create(
        <Row align="middle">
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
        </Row>
      )
      .toJSON();
    const bottom = renderer
      .create(
        <Row align="bottom">
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
        </Row>
      )
      .toJSON();
    expect(top).toMatchSnapshot();
    expect(middle).toMatchSnapshot();
    expect(bottom).toMatchSnapshot();
  });

  it("测试 Row 组件 gutter 参数", () => {
    const numberType = renderer.create(numberTypeComponent).toJSON();
    const objectType = renderer.create(objectTypeComponent).toJSON();
    const ArrayNumberType = renderer.create(ArrayNumberComponent).toJSON();
    const ArrayObjectType = renderer.create(ArrayObjectComponent).toJSON();
    expect(numberType).toMatchSnapshot();
    expect(objectType).toMatchSnapshot();
    expect(ArrayNumberType).toMatchSnapshot();
    expect(ArrayObjectType).toMatchSnapshot();
  });

  screenList.forEach((width) => {
    it(`测试 Row 组件 gutter 参数, window.innerWidth 为 ${width}`, () => {
      resizeWindow(width);
      const row = renderer.create(ArrayObjectComponent).toJSON();
      expect(row).toMatchSnapshot();
    });
  });

  it(`测试 Row 组件 gutter 参数, gutter 为 undefined`, () => {
    const row = renderer
      .create(
        <Row
          gutter={{
            xs: undefined,
            sm: undefined,
            md: undefined,
            lg: undefined,
            xl: undefined,
            xxl: undefined,
          }}
        >
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
          <Col span={6}>col-6</Col>
        </Row>
      )
      .toJSON();
    expect(row).toMatchSnapshot();
  });
});
