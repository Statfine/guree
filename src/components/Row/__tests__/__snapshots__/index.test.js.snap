// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row 正确导出 Row 组件 1`] = `
<div
  className="guree-row"
  style={Object {}}
/>
`;

exports[`Row 测试 Row 组件 align 参数 1`] = `
<div
  className="guree-row guree-row-top"
  style={Object {}}
>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
</div>
`;

exports[`Row 测试 Row 组件 align 参数 2`] = `
<div
  className="guree-row guree-row-middle"
  style={Object {}}
>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
</div>
`;

exports[`Row 测试 Row 组件 align 参数 3`] = `
<div
  className="guree-row guree-row-bottom"
  style={Object {}}
>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
</div>
`;

exports[`Row 测试 Row 组件 gutter 参数 1`] = `
<div
  className="guree-row"
  style={
    Object {
      "marginLeft": -8,
      "marginRight": -8,
    }
  }
>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingLeft": 8,
        "paddingRight": 8,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingLeft": 8,
        "paddingRight": 8,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingLeft": 8,
        "paddingRight": 8,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingLeft": 8,
        "paddingRight": 8,
      }
    }
  >
    col-6
  </div>
</div>
`;

exports[`Row 测试 Row 组件 gutter 参数 2`] = `
<div
  className="guree-row"
  style={
    Object {
      "marginLeft": -16,
      "marginRight": -16,
    }
  }
>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
  >
    col-6
  </div>
</div>
`;

exports[`Row 测试 Row 组件 gutter 参数 3`] = `
<div
  className="guree-row"
  style={
    Object {
      "marginBottom": 12,
      "marginLeft": -8,
      "marginRight": -8,
      "marginTop": -12,
    }
  }
>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 12,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 12,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 12,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 12,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 12,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 12,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 12,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 12,
      }
    }
  >
    col-6
  </div>
</div>
`;

exports[`Row 测试 Row 组件 gutter 参数 4`] = `
<div
  className="guree-row"
  style={
    Object {
      "marginBottom": 10,
      "marginLeft": -4,
      "marginRight": -4,
      "marginTop": -10,
    }
  }
>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 10,
        "paddingLeft": 4,
        "paddingRight": 4,
        "paddingTop": 10,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 10,
        "paddingLeft": 4,
        "paddingRight": 4,
        "paddingTop": 10,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 10,
        "paddingLeft": 4,
        "paddingRight": 4,
        "paddingTop": 10,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 10,
        "paddingLeft": 4,
        "paddingRight": 4,
        "paddingTop": 10,
      }
    }
  >
    col-6
  </div>
</div>
`;

exports[`Row 测试 Row 组件 gutter 参数, gutter 为 undefined 1`] = `
<div
  className="guree-row"
  style={Object {}}
>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
</div>
`;

exports[`Row 测试 Row 组件 gutter 参数, window.innerWidth 为 575 1`] = `
<div
  className="guree-row"
  style={
    Object {
      "marginBottom": 7,
      "marginLeft": -1,
      "marginRight": -1,
      "marginTop": -7,
    }
  }
>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 7,
        "paddingLeft": 1,
        "paddingRight": 1,
        "paddingTop": 7,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 7,
        "paddingLeft": 1,
        "paddingRight": 1,
        "paddingTop": 7,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 7,
        "paddingLeft": 1,
        "paddingRight": 1,
        "paddingTop": 7,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 7,
        "paddingLeft": 1,
        "paddingRight": 1,
        "paddingTop": 7,
      }
    }
  >
    col-6
  </div>
</div>
`;

exports[`Row 测试 Row 组件 gutter 参数, window.innerWidth 为 576 1`] = `
<div
  className="guree-row"
  style={
    Object {
      "marginBottom": 8,
      "marginLeft": -2,
      "marginRight": -2,
      "marginTop": -8,
    }
  }
>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 8,
        "paddingLeft": 2,
        "paddingRight": 2,
        "paddingTop": 8,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 8,
        "paddingLeft": 2,
        "paddingRight": 2,
        "paddingTop": 8,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 8,
        "paddingLeft": 2,
        "paddingRight": 2,
        "paddingTop": 8,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 8,
        "paddingLeft": 2,
        "paddingRight": 2,
        "paddingTop": 8,
      }
    }
  >
    col-6
  </div>
</div>
`;

exports[`Row 测试 Row 组件 gutter 参数, window.innerWidth 为 768 1`] = `
<div
  className="guree-row"
  style={
    Object {
      "marginBottom": 9,
      "marginLeft": -3,
      "marginRight": -3,
      "marginTop": -9,
    }
  }
>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 9,
        "paddingLeft": 3,
        "paddingRight": 3,
        "paddingTop": 9,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 9,
        "paddingLeft": 3,
        "paddingRight": 3,
        "paddingTop": 9,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 9,
        "paddingLeft": 3,
        "paddingRight": 3,
        "paddingTop": 9,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 9,
        "paddingLeft": 3,
        "paddingRight": 3,
        "paddingTop": 9,
      }
    }
  >
    col-6
  </div>
</div>
`;

exports[`Row 测试 Row 组件 gutter 参数, window.innerWidth 为 992 1`] = `
<div
  className="guree-row"
  style={
    Object {
      "marginBottom": 10,
      "marginLeft": -4,
      "marginRight": -4,
      "marginTop": -10,
    }
  }
>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 10,
        "paddingLeft": 4,
        "paddingRight": 4,
        "paddingTop": 10,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 10,
        "paddingLeft": 4,
        "paddingRight": 4,
        "paddingTop": 10,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 10,
        "paddingLeft": 4,
        "paddingRight": 4,
        "paddingTop": 10,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 10,
        "paddingLeft": 4,
        "paddingRight": 4,
        "paddingTop": 10,
      }
    }
  >
    col-6
  </div>
</div>
`;

exports[`Row 测试 Row 组件 gutter 参数, window.innerWidth 为 1200 1`] = `
<div
  className="guree-row"
  style={
    Object {
      "marginBottom": 11,
      "marginLeft": -5,
      "marginRight": -5,
      "marginTop": -11,
    }
  }
>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 11,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 11,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 11,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 11,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 11,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 11,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 11,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 11,
      }
    }
  >
    col-6
  </div>
</div>
`;

exports[`Row 测试 Row 组件 gutter 参数, window.innerWidth 为 1600 1`] = `
<div
  className="guree-row"
  style={
    Object {
      "marginBottom": 12,
      "marginLeft": -6,
      "marginRight": -6,
      "marginTop": -12,
    }
  }
>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 12,
        "paddingLeft": 6,
        "paddingRight": 6,
        "paddingTop": 12,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 12,
        "paddingLeft": 6,
        "paddingRight": 6,
        "paddingTop": 12,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 12,
        "paddingLeft": 6,
        "paddingRight": 6,
        "paddingTop": 12,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
        "paddingBottom": 12,
        "paddingLeft": 6,
        "paddingRight": 6,
        "paddingTop": 12,
      }
    }
  >
    col-6
  </div>
</div>
`;

exports[`Row 测试 Row 组件 justify 参数 1`] = `
<div
  className="guree-row guree-row-start"
  style={Object {}}
>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
</div>
`;

exports[`Row 测试 Row 组件 justify 参数 2`] = `
<div
  className="guree-row guree-row-end"
  style={Object {}}
>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
</div>
`;

exports[`Row 测试 Row 组件 justify 参数 3`] = `
<div
  className="guree-row guree-row-center"
  style={Object {}}
>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
</div>
`;

exports[`Row 测试 Row 组件 justify 参数 4`] = `
<div
  className="guree-row guree-row-space-around"
  style={Object {}}
>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
</div>
`;

exports[`Row 测试 Row 组件 justify 参数 5`] = `
<div
  className="guree-row guree-row-space-between"
  style={Object {}}
>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
  <div
    className="guree-col guree-col-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  >
    col-6
  </div>
</div>
`;
